ProcessLib/BoundaryCondition/PressureBoundaryConditionLocalAssembler.h

```diff
@@ -0,0 +1,174 @@
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#pragma once
+
+#include "MathLib/LinAlg/Eigen/EigenMapTools.h"
+#include "NumLib/DOF/DOFTableUtil.h"
+#include "ProcessLib/Parameter/Parameter.h"
+
+#include "GenericNaturalBoundaryConditionLocalAssembler.h"
+
+namespace ProcessLib
+{
+namespace PressureBoundaryCondition
+{
+template <typename ShapeMatricesTypeDisplacement, int GlobalDim, int NPoints>
+struct IntegrationPointData final
+{
+    IntegrationPointData(
+        typename ShapeMatricesTypeDisplacement::template VectorType<
+            NPoints * GlobalDim> const& Nu_times_n_,
+        double const integration_weight_)
+        : Nu_times_n(Nu_times_n_), integration_weight(integration_weight_)
+    {
+    }
+
+    // Shape matrix (for displacement) times element's normal vector.
+    typename ShapeMatricesTypeDisplacement::template VectorType<
+        NPoints * GlobalDim> const Nu_times_n;
+
+    double const integration_weight;
+};
+
+class PressureBoundaryConditionLocalAssemblerInterface
+{
+public:
+    virtual void assemble(
+        std::size_t const id,
+        NumLib::LocalToGlobalIndexMap const& dof_table_boundary, double const t,
+        const GlobalVector& /*x*/, GlobalMatrix& /*K*/, GlobalVector& b) = 0;
+    virtual ~PressureBoundaryConditionLocalAssemblerInterface() = default;
+};
+
+template <typename ShapeFunctionDisplacement, typename IntegrationMethod,
+          unsigned GlobalDim>
+class PressureBoundaryConditionLocalAssembler final
+    : public PressureBoundaryConditionLocalAssemblerInterface
+{
+public:
+    using ShapeMatricesTypeDisplacement =
+        ShapeMatrixPolicyType<ShapeFunctionDisplacement, GlobalDim>;
+    using GlobalDimVectorType =
+        typename ShapeMatrixPolicyType<ShapeFunctionDisplacement,
+                                       GlobalDim>::GlobalDimVectorType;
+
+    PressureBoundaryConditionLocalAssembler(MeshLib::Element const& e,
+                                            std::size_t const local_matrix_size,
+                                            bool is_axially_symmetric,
+                                            unsigned const integration_order,
+                                            Parameter<double> const& pressure)
+        : _integration_method(integration_order), _pressure(pressure)
+    {
+        _local_rhs.setZero(local_matrix_size);
+
+        unsigned const n_integration_points =
+            _integration_method.getNumberOfPoints();
+
+        _ip_data.reserve(n_integration_points);
+
+        auto const shape_matrices_u =
+            initShapeMatrices<ShapeFunctionDisplacement,
+                              ShapeMatricesTypeDisplacement, IntegrationMethod,
+                              GlobalDim>(e, is_axially_symmetric,
+                                         _integration_method);
+
+        GlobalDimVectorType element_normal(GlobalDim);
+
+        // TODO Extend to rotated 2d meshes and line elements.
```

Comment:
Extend to axisymmetry? That case could be included in  ``shape_matrices_u[ip].integralMeasure``.